includes/class-newspack-newsletters.php

```diff
@@ -1313,5 +1181,39 @@ public static function is_newsletter_sent( $post_id ) {
 		}
 		return false;
 	}
+
+	/**
+	 * Fix the post status of a newsletter. Ensures a newsletter is 'private' if
+	 * the 'is_public' is not found or false.
+	 *
+	 * @param WP_Post $post The post object.
+	 */
+	public static function fix_public_status( $post ) {
+		// Only run if it's a newsletter post.
+		if ( ! self::validate_newsletter_id( $post->ID ) ) {
+			return;
+		}
+		$is_public = (bool) get_post_meta( $post->ID, 'is_public', true );
+		if ( 'publish' === $post->post_status && ! $is_public ) {
+			wp_update_post(
+				[
+					'ID'          => $post->ID,
+					'post_status' => 'private',
+				],
+				false,
+				false
+			);
+			// Force a page refresh on the front-end.
+			if ( ! is_admin() ) {
+				add_action(
+					'wp_footer',
+					function() {
+						header( 'Refresh:0' );
```

Comment:
Maintaining the progressive approach, it's possible to `exit` and refresh as soon as we identify the newsletter to fix through `the_post()`.  This would happen before we print newsletter data into the HTML but would also result in multiple refreshes if the loop has more than 1 newsletter to fix. The reason I decided to hook the refresh to `wp_footer` is so that a single page render fixes as many newsletters as possible and perform a single refresh.

Reply:
That makes sense. Would it be possible to perform the fix without any refreshes, though? Before the query results are sent to the client?

Reply:
Some queries (homepage posts block) are made in the middle of the page.

If we're not using `the_post` hook, which I think we should for practicality, I'd rather not use the progressive approach and do a complete one-time migration.

Reply:
75693d9deacd2ca73983fc00c39181945ec7f6e1 implements the change to exit and refresh immediately, preventing private newsletter data to be printed on the page.

Reply:
Works like a charm!